base/server/src/main/java/com/netscape/cmscore/authentication/ChallengePhraseAuthentication.java

```diff
@@ -314,15 +308,12 @@ private boolean compareChallengePassword(CertRecord record, String pwd)
             return false;
         }
 
-        if (!challengeString.equals(hashpwd)) {
-            return false;
-
-            /*
-             logger.error("ChallengePhraseAuthentication: Incorrect challenge phrase password used for revocation");
-             throw new EInvalidCredentials();
-             */
-        } else
+        if (challengeString.equals(hashpwd)) {
+            logger.debug("ChallengePhraseAuthentication: Correct challenge phrase password used for revocation");
             return true;
+        }
+        logger.error("ChallengePhraseAuthentication: Incorrect challenge phrase password used for revocation");
+        return false;
```

Comment:
This is fine, but in general it might make more sense to check all errors first, then return `true` at the end if there's no errors:

<pre>
if (error) return false;
...
if (error) return false;
return true;
</pre>

Reply:
Yeah it was weighing up which was harder to read:

- A flow that goes `false`, `false`, `false` ... `true`, `false`
- A flow that has `if not condition; return false`

I don't think there is a clear winner, so went with avoiding the former.

Reply:
Thanks for the update! Just another supporting argument: in case we need to check an additional error condition, it would be easier if we have the `true` at the end.